### PR TITLE
Proceed with model training even if saving preprocessed data fails.

### PR DESCRIPTION
**Pull request by justinxzhao**
Running experiments in parallel, sometimes I run into an error where saving the data as a 
parquet file fails (perhaps there's a duplicate or pre-existing file).

Perhaps it's reasonable to catch the exception and log a warning, simply skipping saving preprocessed data, and continuing to proceed to training, rather than crashing.

### CODE REVIEW

File: ludwig/data/preprocessing.py [hunk 1]
```diff
@@ -1791,17 +1791,20 @@ def _preprocess_file_for_training(
...
         logger.debug("writing split file")
         splits_df = concatenate_splits(training_data, validation_data, test_data, backend)
         split_fp = get_split_path(dataset or training_set)
-        backend.df_engine.to_parquet(splits_df, split_fp, index=True)
+        try:
+            backend.df_engine.to_parquet(splits_df, split_fp, index=True)
+        except Exception as e:
+            logger.warning(
+                f"Encountered error: '{e}' while writing data to parquet during saving processed input. "
+                "Skipping saving processed input."
```
Comment (arnavgarg1) [inline ludwig/data/preprocessing.py:1806]: nit: `preprocessed` instead of `processed`?

File: ludwig/data/preprocessing.py [hunk 2]
```diff
@@ -1791,17 +1791,20 @@ def _preprocess_file_for_training(
     else:
         raise ValueError("either data or data_train have to be not None")
 
-    # print("backend", backend)
-    # print("data", backend.df_engine.compute(data))
-
```
Comment (arnavgarg1) [inline ludwig/data/preprocessing.py:1796]: wow can't believe this made it into Ludwig, good catch

**Review by arnavgarg1 — APPROVED** [review]

File: ludwig/data/preprocessing.py (hunk 1 above)
Comment (justinxzhao) [reply ludwig/data/preprocessing.py]: Done.

File: ludwig/data/preprocessing.py (hunk 2 above)
Comment (arnavgarg1) [reply ludwig/data/preprocessing.py:1796]: Definitely does :)

**Pull request merged**